tasks/determine-image-expiration-task.yaml

```diff
@@ -56,7 +56,7 @@ spec:
       echo "Target branch: ${PIPELINE_TARGET_BRANCH}"
 
       is_retest_tagged_build="no"
-      if [[ "${PIPELINE_EVENT_TYPE}" == "retest-comment" ]]; then
+      if [[ "${PIPELINE_EVENT_TYPE}" =~ ^(re)?test-(all-)?comment$ ]]; then
```

Comment:
Thinking about simplifications, I think this task could be rewritten as
```bash
#!/usr/bin/env bash
set -euo pipefail

if [[ "${SOURCE_BRANCH}" == "${TARGET_BRANCH}" && -n "$(git tag --points-at | grep -vF -- '-nightly-')" ]]; then
  # The condition above reads as: this is not a PR and there's a git tag which is not nightly.
  # We don't check for `refs/tags/*` in SOURCE|TARGET_BRANCH because it won't be there when the build is re-triggered with /test or /retest comment.

  echo "This is a tagged build and not a nightly. The images won't expire. IMAGE_EXPIRES_AFTER will be empty."
  IMAGE_EXPIRES_AFTER=""
else
  IMAGE_EXPIRES_AFTER="$(params.DEFAULT_IMAGE_EXPIRES_AFTER)"
fi

echo -n "${IMAGE_EXPIRES_AFTER}" | tee "$(results.IMAGE_EXPIRES_AFTER.path)"
```

WDYT?

Reply:
Well, I think it's a mess today. You don't have to fix it if you don't want to. I'll follow up with my PR after you land this change.